### ISSUE TITLE
ModuleNotFoundError: No module named 'testfixtures'

### ISSUE THREAD

**Issue by bjpcjp** [RaRe-Technologies/gensim#2804]
#### Problem description

* running the following code in `/docs/notebooks/howtos/run_doc2vec_imdb.ipynb`:
`from gensim.test.test_doc2vec import ConcatenatedDoc2Vec`

![Selection_013](https://user-images.githubusercontent.com/79341/80056798-aabdca80-84ea-11ea-8cf9-c2d60b6ecd0d.png)

* A quick review of `/gensim/test/test_doc2vec.py` shows the offending import on line#21. I've tried replacing the line with `from fixtures import log_capture`, but with no improvement.

#### Steps/code/corpus to reproduce

run `run_doc2vec_imdb.ipynb`

#### Versions

Linux-4.15.0-91-generic-x86_64-with-Ubuntu-18.04-bionic
Python 3.6.9 (default, Jul  3 2019, 15:36:16) 
[GCC 5.4.0 20160609]
NumPy 1.18.3
SciPy 1.4.1
gensim 3.8.1
FAST_VERSION 1

**Comment by gojomo** [comment]
It should be sufficient to install the `textfixtures` library: https://pypi.org/project/testfixtures/

**Comment by piskvorky** [comment]
@gojomo @mpenkov looks like some missing installation instructions. How come this notebook passes theour automated doc tests?

**Comment by bjpcjp** [comment]
Fix confirmed via `pip install textfixtures --user`. (Could have sworn that I did this previously.) TY!

**Comment by gojomo** [comment]
I see that this `/docs/notebooks/howtos/run_doc2vec_imdb.ipynb` file isn't in source control; I presume it's copied there from `/docs/src/auto_examples/howtos/run_doc2vec_imdb.ipynb`, which may just be auto-built from `/docs/src/auto_examples/howtos/run_doc2vec_imdb.py` (and thus maybe the `.ipynb` variant shouldn't be in source-control at all?).

My guess would be whatever build-test cycle that this goes through has extra unit-test-supporting packages installed that aren't general prerequisites for `gensim`. This advanced-demo notebook makes use of a utility class from a testing-focused module, hence it succeeds there but not in general user setups (unless they've previously brought in test-config only libraries). 

I'm still not a fan of the extra indirection introduced by the `auto-examples` Sphinx-driven notebooks in a `.py`-format that turns Python comments into notebook cells. Both the run duration of some advanced notebooks – the original doc2vec-IMDB version took an hour or two to run – and the useful instructional technique of including some cells that are intended to error are poorly matched to fully-automatic, every-build testing.

**Comment by mpenkov** [comment]
> @gojomo @mpenkov looks like some missing installation instructions / dependencies. How come this notebook passes our automated doc tests?

We install testfixtures [here](https://github.com/RaRe-Technologies/gensim/blob/d692b9d4cf13c53549406905f4f8b0b4d307a9cf/setup.py#L303).

**Comment by piskvorky** [comment]
Definitely. And if possible, let's link to what `gensim[docs]` entails, so people who didn't use `pip` (or who have other setups / workflows altogether) can install those dependencies manually.

**Comment by mpenkov** [comment]
Where would you show this information? How would you maintain it (ensure that it is correct)?

Perhaps a single separate '[tutorials]' extra would be a better idea?

**Comment by piskvorky** [comment]
> Where would you show this information?

Inside the notebook – either as a comment, or as an explicit `!pip install XYZ==a.b.c` code cell. Comment is probably easier.

> How would you maintain it (ensure that it is correct)?

Not sure. `[tutorials]` isn't fundamentally different from `[docs]`, both seem an overkill for any one notebook.

**Comment by 02abhay** [comment]
I am getting this type of error can anybody help me to resolve this issue??

  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'projectmgmadmin_annotator'
